lichen_cli/src/main.rs

```diff
@@ -200,9 +201,40 @@ fn main() -> color_eyre::Result<()> {
     set_colors_enabled(true);
 
     let euid = unsafe { geteuid() };
-    if euid != 0 {
-        bail!("lichen must be run as root. Re-run with sudo")
-    }
+    ensure!(euid == 0, "lichen must be run as root. Re-run with sudo.");
+
+    let partition_detection_warning = indoc! {"
+        The installer currently does not attempt to detect if there is a file system
+        on detected ESP (and XBOOTLDR) partitions.
+        
+        Please ensure that the EFI system partition (ESP) and the Linux extended boot
+        (XBOOTLDR) partition are both formatted as FAT32.
+        
+        It may be a good idea to check this in gparted (or fdisk) now:
+        - The EFI system partition (>=256MiB) should have the flag 'esp' in gparted
+          - This corresponds to type 1 in fdisk.
+        - The Linux extended boot partition (storing kernels and initrds, 4GiB)
+          should have the flag 'bls_boot' in gparted
+          - This corresponds to type 142 in fdisk.
+        
+        In addition, adding, formatting, and mounting a separate /home partition is
+        not handled by this installer yet, but it is possible to create and format a
+        separate /home partition in gparted, and then enable it in the new Serpent OS
+        '/etc/fstab' file, once the installer finishes. Remember to move the new
+        /home/${USER} directory created by the installer to the new /home partition.
```

Comment:
Idea: How about moving the gparted hints to after installation has succeeded?

```suggestion
        In addition, adding, formatting, and mounting a separate /home partition is
        not handled by this installer yet - instructions will be printed after successful installation.
```

Reply:
We could add an extra reminder? Note that the problem here is that users can't downsize xfs root partitions. So it is probably wiser to ask users to reserve a partition for /home up front?

Reply:
Good point. Not sure about the reminder, whatever you think is best.

Reply:
How about now?